srvutil/metrics.go

```diff
@@ -56,41 +80,50 @@ func (w *hijackableRecorder) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	return w.httpRecorder.ResponseWriter.(http.Hijacker).Hijack()
 }
 
-func newHTTPRecorder(w http.ResponseWriter) loggableHTTPRecorder {
-	recorder := httpRecorder{ResponseWriter: w}
+func newHTTPRecorder(w http.ResponseWriter, bodyLogPredicate BodyLogPredicateFunc) loggableHTTPRecorder {
+	recorder := httpRecorder{ResponseWriter: w, bodyLogPredicate: bodyLogPredicate}
 	if _, ok := w.(http.Hijacker); ok {
 		return &hijackableRecorder{recorder}
 	}
 	return &recorder
 }
 
-// RequestMetricsMiddleware records the time taken to serve a request.
+// RequestMetricsMiddleware records the time taken to serve a request, and logs request and response data.
 // Example tags: statusClass:2xx, statusCode:200
 // Should be added as a middleware after RequestContextMiddleware to benefit from its tags
-func RequestMetricsMiddleware(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
+func RequestMetricsMiddleware(bodyLogPredicate BodyLogPredicateFunc) func(http.Handler) http.Handler {
```

Comment:
This breaks BC. Move this to a `NewRequestMetricsMiddleware` or something, and change this to `var RequestMetricsMiddleware = NewRequestMetricsMiddleware(LogErrorBody)`

Reply:
Fixed.